File: Leetcode/240-Search-a-2D-Matrix-II/SearchA2DMatrixII.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        if (matrix.size() == 0 || matrix[0].size() == 0) {
            return false;
        }
        int i =0;
        int j = matrix[0].size()-1;
        while(i < matrix.size() && j>=0) {
            int temp = matrix[i][j];
            if (temp == target) {
                return true;
            } else if (temp > target) {
                j--;
            } else {
                i++;
            }
        }
        return false;
    }
};

File: Leetcode/240-Search-a-2D-Matrix-II/SearchA2DMatrixII.go
func searchMatrix(matrix [][]int, target int) bool {
	if matrix == nil || len(matrix) == 0 || len(matrix[0]) == 0 {
		return false
	}
	i := 0
	j := len(matrix[0]) - 1
	length := len(matrix)
	for i < length && j >= 0 {
		temp := matrix[i][j]
		if temp == target {
			return true
		}
		if temp > target {
			j--
		} else {
			i++
		}
	}
	return false
}
